src/ProjectOrigin.RequestProcessor/Models/RequestId.cs

```diff
@@ -0,0 +1,21 @@
+namespace ProjectOrigin.RequestProcessor.Models;
+
+public record RequestId(byte[] RequestHash)
+{
+    public virtual bool Equals(RequestId? other)
+    {
+        if (other is null) return false;
+
+        return RequestHash.SequenceEqual(other!.RequestHash);
```

Comment:
This one is wild - other! - is it not :-)

Reply:
If i add ! at the end of other, then i tell the compile that it will have a value, but i want to perform a null check here.

Since RequestId? where the ? denotes that the value is nullable.